lib/components/form/advanced-settings-panel.tsx

```diff
@@ -0,0 +1,288 @@
+
+  const intl = useIntl()
+  const closeButtonText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.closeAdvancedPreferences'
+  })
+  const headerText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.advancedHeader'
+  })
+
+  // @ts-expect-error Context not typed
+  const { ModeIcon } = useContext(ComponentContext)
+
+  const planTrip = () => {
+    alertUserTripPlan(intl, currentQuery, onPlanTripClick, routingQuery)
+    closeAdvancedSettings()
+  }
+
+  const closeAndAnnounce = () => {
+    closeAdvancedSettings()
+    toast.success(intl.formatMessage({ id: 'actions.user.preferencesSaved' }))
+  }
+
+  const processSettings = (settings: ModeSetting[]) =>
+    settings.map(
+      pipe(
+        populateSettingWithIcon(ModeIcon),
+        populateSettingWithValue(modeSettingValues),
+        addCustomSettingLabels(intl)
+      )
+    )
+
+  const globalSettings = modeSettingDefinitions.filter((x) => !x.applicableMode)
+  const processedGlobalSettings = processSettings(globalSettings)
+
+  const globalSettingsComponents = processedGlobalSettings.map(
+    (setting: ModeSetting) => (
+      <ModeSettingRenderer
+        key={setting.key}
+        onChange={onSettingsUpdate(setQueryParam)}
+        setting={setting}
+      />
+    )
+  )
+
+  const processedModeSettings = processSettings(modeSettingDefinitions)
+  const processedModeButtons = modeButtonOptions.map(
+    pipe(
+      addModeButtonIcon(ModeIcon),
+      addSettingsToButton(processedModeSettings),
+      setModeButtonEnabled(enabledModeButtons)
+    )
+  )
+
+  return (
+    <PanelOverlay className="advanced-settings" ref={innerRef}>
+      <HeaderContainer>
+        <PageTitle title={headerText} />
+        <h1 className="header-text">{headerText}</h1>
+        <CloseButton
+          aria-label={closeButtonText}
+          onClick={closeAndAnnounce}
+          title={closeButtonText}
+        >
+          <Close size={22} />
+        </CloseButton>
+      </HeaderContainer>
+      <DtSelectorContainer>
+        <DateTimeModal />
+      </DtSelectorContainer>
+      {processedGlobalSettings.length > 0 && (
+        <>
+          <Subheader>
+            <FormattedMessage id="components.BatchSearchScreen.tripOptions" />
+          </Subheader>
+          <GlobalSettingsContainer className="global-settings-container">
+            {globalSettingsComponents}
+          </GlobalSettingsContainer>
+        </>
+      )}
+      <Subheader>
+        <FormattedMessage id="components.BatchSearchScreen.modeOptions" />
+      </Subheader>
+      <AdvancedModeSubsettingsContainer
+        accentColor={accentColor}
+        fillModeIcons
+        label="test"
+        modeButtons={processedModeButtons}
+        onSettingsUpdate={onSettingsUpdate(setQueryParam)}
+        onToggleModeButton={setModeButton(
+          enabledModeButtons,
+          onSettingsUpdate(setQueryParam)
+        )}
+      />
+      <ButtonContainer>
```

Comment:
Two buttons at the bottom is confusing. Should be one save button for clarity

Reply:
Why is it confusing if both of the buttons communicate what they do? It seems like from this page there are two options on how to proceed (returning to trip form, or pulling the trigger on plan trip) and having them both easily available makes that clear

Reply:
It's a second location for the plan trip button. That's confusing and lacks internal consistency https://www.nngroup.com/articles/ten-usability-heuristics/ #4. Moving the pane next to the other pane should solve this problem even better.